core/src/main/java/org/elasticsearch/index/query/functionscore/ScoreFunctionsRegistry.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query.functionscore;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.index.query.QueryParseContext;
+
+import java.util.Map;
+
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * Registry of all {@link ScoreFunctionParser}s.
+ */
+public class ScoreFunctionsRegistry {
+    private final Map<String, Tuple<ParseField, ScoreFunctionParser<?>>> scoreFunctionParsers;
+
+    public ScoreFunctionsRegistry(Map<String, Tuple<ParseField, ScoreFunctionParser<?>>> scoreFunctionParsers) {
+        this.scoreFunctionParsers = unmodifiableMap(scoreFunctionParsers);
+    }
+
+    /**
+     * Lookup a {@linkplain ScoreFunctionParser} by name, checking that the name matches in the current parse context. It might, for
+     * example, be a deprecated name and the context might be in strict mode.
+     */
+    public ScoreFunctionParser<?> lookup(String name, QueryParseContext context) {
```

Comment:
Does it make sense to pass in the ParseFieldMatcher and the XContentLocation instead of the whole QueryParseContext? I would love for this to be aligned with IndicesQueriesRegistry (we can also go and change it if this other registry is better)